src/auth.ts

```diff
@@ -4,18 +4,29 @@ import { Constants, getResourceIdFromPath, HTTPMethod, ResourceType } from "./co
 import { IHeaders } from "./queryExecutionContext";
 
 /** @hidden */
-export interface IRequestInfo {
-  [index: string]: any;
+export interface RequestInfo {
   verb: HTTPMethod;
   path: string;
   resourceId: string;
-  resourceType: string;
+  resourceType: ResourceType;
   headers: IHeaders;
 }
 
-export interface ITokenProvider {
-  getToken: (requestInfo: IRequestInfo, callback?: (err: Error, token: string) => void) => Promise<string>;
-}
+export type TokenProvider = (
+  verb: HTTPMethod,
+  path: string,
+  resourceId: string,
+  resourceType: ResourceType,
+  headers: IHeaders,
+  /** The default function for generating header tokens using the masterKey */
+  getAuthorizationTokenUsingMasterKey: (
```

Comment:
why are we passing this helper method in each time, not not any of the other helper methods? Shouldn't we just export all these helper methods from a library, and not pass them all in as arguments? This smells weird to me.

Reply:
Agreed. I'll revisit. This was how the previous token provider worked.

Reply:
Changed it to `setAuthorizationTokenHeaderUsingMasterKey` to reflect that it is modifying the header object and exported it from src/index